### DISCUSSION TITLE
Validate struct inside struct with custom functions

### DISCUSSION THREAD

**Discussion by lcucurachi** [go-playground/validator#1137] [Q&A]
Hi everyone, as per title I'm trying to validate structs inside another struct with custom function but none of the solutions or guidelines I have found online have worked for me. I am not sure why.

Say I have:
```
type MainStruct struct {
Val1 *FirstStruct
Val2 *SecondStruct
}
```

I want to be able to validate each struct (if not nil) in a custom function. Does anyone know how to do this please?

**Comment by deankarn** [comment]
@LoKKeR96 in the documentation you may do so by registering as custom struct validation.
